chef/cookbooks/network/recipes/default.rb

```diff
@@ -398,32 +398,49 @@ def get_datapath_id_for_ovsbridge(bridge)
 
   # Do some MTU checks/configuration for the parent of the vlan nic
   if nic.is_a?(Nic::Vlan)
-    # 1) validate that the parent of vlan nic is not used by a network directly
-    # and the requested mtu for the parent is lower than the vlan nic mtu.
-    # That would not work and setting the mtu on the vlan would fail.
-    networks_using_parent = if_mapping.select { |net, ifaces| ifaces.include? nic.parent }.keys
-    networks_using_parent.each do |net_name|
-      net = Barclamp::Inventory.get_network_by_type(node, net_name)
-      unless net.use_vlan
-        nic_parent = Nic.new nic.parent
-        if nic_parent.mtu.to_i < ifs[nic.name]["mtu"].to_i
-          msg = "#{nic.name} wants mtu #{ifs[nic.name]["mtu"]} but network #{net_name} " \
-                "using the parent nic #{nic.parent} wants a lower mtu #{net.mtu}. " \
-                "This network mtu configuration is invalid."
+    if nic.parent
+      parent_nic = Nic.new(nic.parent)
+      if parent_nic.bond?
+        # parent is a bond
+        if parent_nic.key?("mtu") && parent_nic["mtu"] < ifs[nic.name]["mtu"].to_i
```

Comment:
Indeed on the first part, it will have a mtu function, not a key, I'll fix this.

And indeed it seems that on the second part, the mtu function will read from /sys/class/net/DEVICE/mtu which will always be present, even with no mtu set on our side, so this would always be raised.

Will change!